sequencing/ATACSeq/jobSubmission/1_batchRunPreAnalysis.sh

```diff
@@ -0,0 +1,159 @@
+## DESCRIPTION: This script performs the first step in the ATAC-seq pipeline: pre-analysis. This                      ||
+## includes quality control of raw reads (FASTQC), trimming the reads (TRIM), aligning reads to                       ||
+## reference genome (ALIGN) and calculate ENCODEQC metrics (ENCODE).                                                  ||
+##                                                                                                                    ||
+## REQUIRES:                                                                                                          ||
+## - File in ${METADIR}/samples.txt that lists sample names.                                                          ||
+## ===================================================================================================================##
+
+## ============ ##
+##    SET UP    ##
+## ============ ##
+
+## print start date and time when script runs
+echo Job started on:
+date -u
+
+## needs to be executed from the scripts folder
+echo "Changing Folder to: " $SLURM_SUBMIT_DIR
+cd $SLURM_SUBMIT_DIR || exit 1
+
+## load config file provided on command line related to the specified project
+echo "Loading config file for project: " $1
+export PROJECT=$1
+source ./ATACSeq/config/config.txt 
+
+## check directories
+echo "Project directory is: " $DATADIR 
+echo 'Script is running from directory: ' ${SCRIPTDIR}
+
+##check array specified and exit if not
+if [[ ${SLURM_ARRAY_TASK_ID} == '' ]]
+then 
+    { echo "Job does not appear to be an array. Please specify --array on the command line." ; exit 1; }
+fi
+
+## check step method matches required options and exit if not
+if [[ ! $2 =~ "FASTQC" ]] && [[ ! $2 =~ "TRIM" ]] && [[ ! $2 =~ "ALIGN" ]] && [[ ! $2 =~ "ENCODE" ]] &&[[ ! $2 == '' ]];
+then 
+    { echo "Unknown step specified. Please use FASTQC, TRIM, ALIGN, ENCODE or some combination of this as a single string (i.e. FASTQC,TRIM)" ; exit 1; }            
+fi
+
+## ============ ##
+##    STEPS     ##
+## ============ ##
+
+## check if file containing list of sample IDs exists and if so:
+if test -f ${METADIR}/samples.txt;
+then 
+    ## create an array from the file
+    mapfile -t SAMPLEIDS < ${METADIR}/samples.txt 
+    echo "Number of sample IDs found:"" ""${#SAMPLEIDS[@]}"""
+    
+    ## Sample(s) specified in by array number(s) from command line
+    sampleID=${SAMPLEIDS[${SLURM_ARRAY_TASK_ID}]}
+    dirs=( ${RAWDATADIR} ${RAWDATADIR_B1} ${RAWDATADIR_B2} )
+    #toProcess=($(find ${RAWDATADIR} -maxdepth 1 -name ${SAMPLEIDS[${SLURM_ARRAY_TASK_ID}]}'*'))
+    toProcess=($(find "${dirs[@]}" -maxdepth 1 -name ${SAMPLEIDS[${SLURM_ARRAY_TASK_ID}]}'*'))
+    ## sort the toProcess array so that R1 and R2 are consecutive 
+    IFS=$'\n' # need to set this as \n rather than default - a space, \t and then \n - so that elements are expanded using \n as delimiter
+    toProcess=($(sort <<<"${toProcess[*]}")) ## sort so that the first element is R1
```

Comment:
I'm not convinced this is actually sorting the array. This command is parsing the array as a string and so the sort will just output the array as is.

Observe:
```bash
[me@login01 test] $ array=(apple pear banana orange)
[me@login01 test] $ sort <<<"${array[*]}"
apple pear banana orange
```

I found this solution online if this actually matters:
`toProcess=($(printf "%s\n" "${toProcess[@]}" | sort))`
I can't figure out how to get this to work with a regular `echo` however.